### PR TITLE
CHE-1007: add docker and development views for Edit targets

### PR DESCRIPTION
**Pull request by olexii4**
Signed-off-by: Oleksii Orel oorel@codenvy.com
@ashumilova, @azatsarynnyy, @vparfonov, @vitaliy-guliy  
selection_075

### CODE REVIEW

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/CategoryPageRegistry.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+            }
+        }
+    }
+
+    /**
+     * Returns CategoryPage with the specified Category or {@code null} if none.
+     *
+     * @param targetCategory
+     *         the type of the target
+     * @return target or {@code null}
```
Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/CategoryPageRegistry.java]: javadoc says that method returns target but actually it returns CategoryPage

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/CategoryPageRegistry.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            if (this.categoryPageByCategoryMap.containsKey(category)) {
+                Log.warn(this.getClass(), "Category page with category '" + category + "' is already registered.");
+            } else {
+                this.categoryPageByCategoryMap.put(category, page);
+            }
+        }
+    }
+
+    /**
+     * Returns CategoryPage with the specified Category or {@code null} if none.
```
Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/CategoryPageRegistry.java]: `with the specified Category` or `for the specified Category`?
I think the second variant is better

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+     * Sets target name.
+     *
+     * @param targetName
+     */
+    void setTargetName(String targetName);
+
+    /**
+     * Returns value of target name field.
+     *
+     * @return
```
Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: not formatted javadoc

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+     * Sets SSH host value.
+     *
+     * @param host
+     */
+    void setHost(String host);
+
+    /**
+     * Returns value of host field.
+     *
+     * @return
```
Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: the same as https://github.com/eclipse/che/pull/1261#discussion_r63479520

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+     *
+     * @param userName
+     *          user name
+     */
+    void setUserName(String userName);
+
+    /**
+     * Returns value of user name field.
+     *
+     * @return
```
Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: the same as https://github.com/eclipse/che/pull/1261#discussion_r63479520

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+     *
+     * @param password
+     *          password
+     */
+    void setPassword(String password);
+
+    /**
+     * Returns value of password field.
+     *
+     * @return
```
Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: the same as https://github.com/eclipse/che/pull/1261#discussion_r63479520

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java [hunk 7]
```diff
@@ -0,0 +1,676 @@
...
+                dialogFactory.createMessageDialog("Error", machineLocale.targetsViewDeleteError(target.getName()), null).show();
+            }
+        });
+    }
+
+    /**
+     * Subscribes to the websocket channel and starts listening machine status events.
+     *
+     * @param machineName
+     *         mane of the machine to subscribe
```
Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java:566]: mane?
грива? :smile:

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java [hunk 8]
```diff
@@ -12,117 +12,50 @@
 
 import org.eclipse.che.api.machine.shared.dto.recipe.RecipeDescriptor;
 
+import javax.validation.constraints.NotNull;
+
 /**
- * Wrapper fot the Recipe.
+ * Wrapper fot the machine.
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java]: Typo

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java [hunk 9]
```diff
@@ -133,60 +87,71 @@ public void execute() {
...
-                        }
+                        selectTarget(preselectTargetName == null ? selectedTarget : targetByName.get(preselectTargetName));
                     }
                 });
 
             }
         });
     }
 
+    private Target createTarget(String targetName, String targetCategory) {
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java:138]: Javadoc?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java [hunk 10]
```diff
@@ -12,117 +12,50 @@
...
-        return connected;
-    }
+    /** Sets status of unsaved changes. */
+    void setDirty(boolean dirty);
 
-    public void setConnected(boolean connected) {
-        this.connected = connected;
-    }
+    /** Restore target if it possible. */
+    boolean onRestore();
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java]: What this method does?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java [hunk 11]
```diff
@@ -12,117 +12,50 @@
...
 
+    /** Update field values. */
+    void onSelect();
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java]: What this method does?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetFactory.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Factory for {@link Target} instances.
+ *
+ * @author Oleksii Orel
+ */
+public abstract class TargetFactory {
+
+    /**
+     * Creates a new command configuration based on the given {@link CommandDto}.
```
Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetFactory.java]: Where is `given CommandDto`?
Seems that this javadoc is unsuitable here

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java [hunk 13]
```diff
@@ -40,201 +44,52 @@
...
 
-        // Perform actions when selecting a target
-        void onTargetSelected(Target target);
-
-        void onTargetNameChanged(String value);
-
-        void onHostChanged(String value);
-
-        void onPortChanged(String value);
+        void updateTargets(String preselectTargetName);
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java]: What does this method do?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentCategoryPresenter.java [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+import org.eclipse.che.ide.extension.machine.client.targets.CategoryPage;
+import org.eclipse.che.ide.extension.machine.client.targets.TargetFactory;
+import org.eclipse.che.ide.extension.machine.client.targets.TargetsView;
+
+/**
+ * Development type page presenter.
+ *
+ * @author Oleksii Orel
+ */
+public class DevelopmentCategoryPresenter implements CategoryPage, DevelopmentView.ActionDelegate {
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentCategoryPresenter.java]: What does development type mean?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java [hunk 15]
```diff
@@ -40,201 +44,52 @@
...
-        void onPasswordChanged(String value);
+        // Perform actions when clicking Close button
+        void onCloseClicked();
 
-        void onSaveClicked();
+        // Perform actions when selecting a target
+        void onTargetSelected(Target target);
 
-        void onCancelClicked();
+        boolean isTargetNameExist(String targetName);
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java]: Delegates are used to delegate user events from view to its presenter. And only for that.
Don't use any other methods, only handle events.

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentMachineTarget.java [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+import org.eclipse.che.ide.extension.machine.client.targets.Target;
+
+import java.util.Objects;
+
+/**
+ * The implementation of {@link Target}.
+ *
+ * @author Oleksii Orel
+ */
+public class DevelopmentMachineTarget implements Target {
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentMachineTarget.java]: What is Development target ?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java (hunk 15 above)
Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java]: +1

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentViewImpl.java [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+
+/**
+ * The implementation of {@link DevelopmentView}.
+ *
+ * @author Oleksii Orel
+ */
+@Singleton
+public class DevelopmentViewImpl implements DevelopmentView {
+
+    private static final SshViewImplUiBinder UI_BINDER = GWT.create(SshViewImplUiBinder.class);
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentViewImpl.java]: SshViewImplUiBinder in DevelopmentViewImpl

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/docker/DockerCategoryPresenter.java [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+        return this.delegate != null ? this.delegate.getMachineByName(machineName) : null;
+    }
+
+    private void updateTargets(String preselectTargetName) {
+        if (this.delegate == null) {
+            return;
+        }
+        this.delegate.updateTargets(preselectTargetName);
+    }
+
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/docker/DockerCategoryPresenter.java:101]: if (delegate != null) {
    delegate.updateTargets(preselectTargetName);
}

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/docker/DockerViewImpl.java [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+        this.setOwner(target.getOwner());
+        this.setType(target.getType());
+        this.setSourceType(target.getSourceType());
+        this.setSourceUrl(target.getSourceUrl());
+    }
+
+    @Override
+    public boolean restoreTargetFields(DockerMachineTarget target) {
+        return this.delegate != null && this.delegate.onRestoreTargetFields(target);
+    }
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/docker/DockerViewImpl.java:110]: It's difficult to understand. Delegates are used to delegate events from view to its presenter.
Here an implementation of
boolean restoreTargetFields(DockerMachineTarget target) 
is used to call a method from the delegate. It looks the method wants to be refactored or removed.

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java [hunk 20]
```diff
@@ -0,0 +1,676 @@
...
+    }
+
+    @Override
+    public String getCategory() {
+        return this.machineLocale.targetsViewCategorySsh();
+    }
+
+    @Override
+    public TargetFactory getTargetFactory() {
+        return targetFactory;
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java]: Use `this` everywhere or nowhere

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java [hunk 21]
```diff
@@ -0,0 +1,676 @@
...
+                                              }
+                                          }, new CancelCallback() {
+                    @Override
+                    public void cancelled() {
+                        updateTargets(null);
+                    }
+                }).show();
+    }
+
+    private void deleteTarget(final Target target) {
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java:504]: A little javadoc?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java (hunk 7 above)
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java:566]: Мається на увазі - ухватити за гриву

Comment (azatsarynnyy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java:566]: о! тепер ясно :horse: 
так би й відразу
thx @vitaliy-guliy for explanation

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshMachineTargetFactory.java [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+ * @author Oleksii Orel
+ */
+public class SshMachineTargetFactory extends TargetFactory {
+    private final MachineLocalizationConstant machineLocale;
+    private final SshView.ActionDelegate delegate;
+    private final SshView view;
+
+    protected SshMachineTargetFactory(@NotNull MachineLocalizationConstant machineLocale,
+                                      @NotNull SshView view,
+                                      @NotNull SshView.ActionDelegate delegate) {
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshMachineTargetFactory.java]: Factory and view action delegate are not friends.

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java [hunk 23]
```diff
@@ -0,0 +1,207 @@
...
+    void setConnectButtonText(String text);
+
+    /**
+     * Focuses and selects all the text in the Name field.
+     */
+    void selectTargetName();
+
+    interface ActionDelegate {
+        //update enable state of buttons
+        void updateButtons();
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: Remove it from ActionDelegate

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java [hunk 24]
```diff
@@ -0,0 +1,207 @@
...
+
+        void onDeleteTarget(Target target);
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: A little javadoc on the following

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshViewImpl.java [hunk 25]
```diff
@@ -0,0 +1,344 @@
...
+        saveButton.setEnabled(enable);
+    }
+
+    @Override
+    public void enableCancelButton(boolean enable) {
+        cancelButton.setEnabled(enable);
+    }
+
+    @Override
+    public boolean restoreTargetFields(SshMachineTarget target) {
```
Comment (vitaliy-guliy) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshViewImpl.java:263]: Following code wants to be in presenter

Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshViewImpl.java:263]: Please, justify. Why do you want to use Button widgets directly in presenter(not setters methods from view)?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java (hunk 24 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java (hunk 23 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshMachineTargetFactory.java (hunk 22 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshMachineTargetFactory.java]: ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java (hunk 21 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java:504]: ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java (hunk 20 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java]: I use it everywhere. Do you see a mistake? Where is it?

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/docker/DockerViewImpl.java (hunk 19 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/docker/DockerViewImpl.java:110]: OK. I will changed it

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/docker/DockerCategoryPresenter.java (hunk 18 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/docker/DockerCategoryPresenter.java:101]: I have another opinion

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentViewImpl.java (hunk 17 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentViewImpl.java]: Typo

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentMachineTarget.java (hunk 16 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentMachineTarget.java]: https://github.com/eclipse/che/blob/ad14bf1d3db2bf939301237fea31fcd72c2dc53b/plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java#L104

Type:docker && Dev:true

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java (hunk 15 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java]: OK

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java (hunk 13 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsView.java]: This method make a  new list of targets.  I have removed it from delegate.

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetFactory.java (hunk 12 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetFactory.java]: Typo

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java (hunk 11 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java]: I have removed it from Target.

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java (hunk 10 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/Target.java]: I have removed it from Target.

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java (hunk 9 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/TargetsPresenter.java:138]: Ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java (hunk 7 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshCategoryPresenter.java:566]: No. Only typo)))

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java (hunk 6 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java (hunk 5 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java (hunk 4 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java (hunk 3 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/ssh/SshView.java]: ok

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/CategoryPageRegistry.java (hunk 2 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/CategoryPageRegistry.java]: ok
for the specified Category

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/CategoryPageRegistry.java (hunk 1 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/CategoryPageRegistry.java]: ok. Typo

File: plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentCategoryPresenter.java (hunk 14 above)
Comment (olexii4) [inline plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/targets/categories/development/DevelopmentCategoryPresenter.java]: Do  you  mean  development  category?
Type:docker && Dev:false

**Comment by ashumilova** [comment]
+

**Comment by azatsarynnyy** [comment]
+

**Comment by vitaliy-guliy** [comment]
+0

**Comment by vparfonov** [comment]
ok

**Pull request merged**